### PR TITLE
[WIP] Upgrade to webpack 2

### PR DESCRIPTION
**Pull request by Benaiah**
**- Summary**

Upgrade to webpack 2 for building

**- Test plan**

TBD; need to make sure we don't have build regressions from webpack 1

**- Description for the changelog**

Upgrade to webpack 2

### CODE REVIEW

File: webpack.dev.js [hunk 1]
```diff
@@ -45,18 +44,19 @@ module.exports = merge.smart(require('./webpack.base.js'), {
       },
     }),
     new webpack.DefinePlugin({
-      NETLIFY_CMS_VERSION: JSON.stringify(require("./package.json").version + "-dev")
+      NETLIFY_CMS_VERSION: JSON.stringify(require("./package.json").version + "-dev"),
     }),
-    new webpack.optimize.OccurenceOrderPlugin(),
     new webpack.HotModuleReplacementPlugin(),
-    new webpack.NoErrorsPlugin(),
-    new ExtractTextPlugin('[name].css', { disable: true }),
```
Comment (erquhart) [inline webpack.dev.js]: Wonder if there's a reason we're using ExtractText for dev.

File: webpack.dev.js [hunk 2]
```diff
@@ -45,18 +44,19 @@ module.exports = merge.smart(require('./webpack.base.js'), {
...
+    new ExtractTextPlugin({
+      filename: '[name].css',
+      disable: true,
+    }),
   ],
   devServer: {
     hot: true,
     contentBase: 'example/',
     historyApiFallback: true,
-    devTool: 'cheap-module-source-map',
```
Comment (erquhart) [inline webpack.dev.js]: No sourcemaps?

Comment (Benaiah) [inline webpack.dev.js]: The sourcemaps are still there as far as I can tell, but I didn't yet find a documented place to move this setting.

Comment (erquhart) [inline webpack.dev.js]: https://webpack.js.org/configuration/devtool/#devtool

Comment (Benaiah) [inline webpack.dev.js]: Fixed in https://github.com/netlify/netlify-cms/pull/451/commits/275519825f534f6427c7e376e52947e1f0de4a27

**Comment by bdougie** [comment]
I am seeing this a https://www.dropbox.com/s/5g27hgzo4dmz0qx/Screenshot%202017-06-15%2008.18.18.png?dl=0

Seems like the css loader is moving css over in the bundle

screenshot 2017-06-15 08 18 18

**Comment by erquhart** [comment]
@bdougie is that from the prod build? I'm not seeing that when running the dev build.

**Comment by erquhart** [comment]
oh yep, prod

**Comment by erquhart** [comment]
Nah, still happening with Kaldi.

**Comment by erquhart** [comment]
Only happening on 127.0.0.1, not localhost.

**Comment by erquhart** [comment]
The issue appears to be the differences between the base and prod rules for css, guessing they're not merging differently now. Excluding node modules and providing the `localIdentName` for css modules has to happen in the prod config now, which really means we should just remove the css portion of the prod config. Both approaches fixed the prod build locally.

**Comment by erquhart** [comment]
Post-mortem: rule objects in a base config aren't extended by the config being merged in - if a rule targets the same files as a rule in the original config, the new rule will be added to the merged config after the original, essentially replacing it.

**Comment by Benaiah** [comment]
@bdougie @erquhart should be fixed in https://github.com/netlify/netlify-cms/pull/451/commits/96874bd956ed15dd73ec3a57769fca1ab911de72.

**Pull request merged**